### PR TITLE
Use DynamoDB for state

### PR DESCRIPTION
**Pull request by prog893**
Depends on https://github.com/baikonur-oss/aws-eden-cli/pull/3

Use DynamoDB for profile retrieval, environment list management.

### CODE REVIEW

File: lambda/main.py [hunk 1]
```diff
@@ -1,73 +1,158 @@
...
+        r = aws_eden_core.methods.create_env(name, image_uri, g.config)
+    except Exception as e:
+        logger.error(e)
         return jsonify({
             'status': 'error',
-            'code':   501
+            'errors': f"Exception caught",
         }), 501
 
+    table = boto3.resource('dynamodb').Table(table_name)
```
Comment (toVersus) [inline lambda/main.py]: I'm not sure which one is preferable, but could you align the way of handling DynamoDB client?

I found this code on the upper side:

https://github.com/baikonur-oss/terraform-aws-lambda-eden-api/blob/c1a1865e10c982d15a09b844a4a5c00742b8ca32/lambda/main.py#L60-L67

Comment (prog893) [reply lambda/main.py]: dynamodb is boto3 DynamoDB client, while `boto3.resource('dynamodb')` is boto3 DynamoDB resource. These are not the same, and it is difficult to name them. 

We can go with dynamodb_client for former, and dynamodb_resource for latter, how does that sound?

Comment (toVersus) [reply lambda/main.py]: Ah, sorry for ignorance. That would be easier to understand :)

Comment (prog893) [reply lambda/main.py]: Will rename here and API parts then.

Comment (prog893) [reply lambda/main.py]: Done for both

File: lambda/main.py [hunk 2]
```diff
@@ -1,73 +1,160 @@
...
+        r = aws_eden_core.methods.delete_env(name, g.config)
+    except Exception as e:
+        logger.error(e)
         return jsonify({
             'status': 'error',
-            'code':   501
+            'errors': f"Exception caught",
         }), 501
 
+    table = boto3.resource('dynamodb').Table(table_name)
```
Comment (toVersus) [inline lambda/main.py]:
```suggestion
    table = dynamodb_resource.Table(table_name)
```

**Review by toVersus — APPROVED** [review]
LGTM

**Pull request merged**